src/integration/prober/prober.go

```diff
@@ -15,19 +16,47 @@ import (
 type Prober struct {
 	timeout time.Duration
 	backoff time.Duration
-	Logger  *log.Logger
+	logger  *log.Logger
+	client  *http.Client
 }
 
 var ErrProbeTimeout = errors.New("probe timed out")
 var errProbeNotOk = errors.New("probe did not return 200 Ok")
 
+type OptionFunc func(p *Prober) error
+
+// WithLogger returns an OptionFunc which tells the Prober to use the specified logger.
+func WithLogger(logger *log.Logger) OptionFunc {
```

Comment:
This is not used

Reply:
Should be, I forgot to make the change 😅 
Good catch!